napari/_qt/qt_dims.py

```diff
@@ -328,3 +330,174 @@ def focus_down(self):
         else:
             index = (displayed.index(self.last_used) - 1) % len(displayed)
             self.last_used = displayed[index]
+
+    def play(self, axis: int = 0, fps: float = 10, range=None, pingpong=False):
+        """
+        Animate (play) axis
+
+        Parameters
+        ----------
+        axis: int
+            Index of axis to play
+        fps: float
+            Frames per second for playback.  Negative values will play in
+            reverse.  fps == 0 will stop the animation. The view is not
+            guaranteed to keep up with the requested fps, and may drop frames
+            at higher fps.
+        range: tuple | list
+            If specified, will constrain animation to loop [first, last] frames
+        pingong: bool
+            If True, movie will loop back and forth. If False (default) movie
+            will return to the first frame after reaching the last frame.
+        Raises
+        ----------
+        IndexError
+            If ``axis`` requested is out of the range of the dims
+        IndexError
+            If ``range`` is provided and out of the range of the dims
+        ValueError
+            If ``range`` is provided and range[0] >= range[1]
+        """
+        # TODO: No access in the GUI yet. Just keybinding.
+
+        # allow only one axis to be playing at a time
+        # if nothing is playing self.stop() will not do anything
+        self.stop()
+        if fps == 0:
+            return
+
+        if axis >= len(self.dims.range):
+            raise IndexError('axis argument out of range')
+        # we want to avoid playing a dimension that does not have a slider
+        # (like X or Y, or a third dimension in volume view.)
+        if not self._displayed_sliders[axis]:
+            return
+
+        self.animation_thread = AnimationThread(
+            self.dims, axis, fps, range, pingpong
```

Comment:
Let's not shadow the built-in `range`. I'm also not in love with `pingpong`. Perhaps we can have `playback_mode={'once', 'loop', 'loop_back_and_forth'}`? (Once being the default.)

Reply:
you know, it's funny, I almost avoided shadowing `range` there, and then saw a [few lines ](https://github.com/napari/napari/blob/0c3f28766eaf37f600306e19c77f74c4a3129f67/napari/_qt/qt_dims.py#L164-L170) in the same file that also shadowed it and thought, "hey I guess if it's only in a small scope, maybe it is a nice word to use..." and so I followed that lead :)
I almost used `boomerang` in honor of australia, instead of `pingpong`... I guess I was feeling whimsical.  `playback_mode` is fine (soo predictable 😉).  Only thing I would add is that I prefer `'loop'` to be the default rather than `'once'`... @sofroniewn ?

Reply:
Oh no... What have I become...? (@stefanv often shoots me down in my whimsical moments, now here I am doing the same... =P)

I'm not strongly opposed to loop being the default, your call!

Reply:
(With Napari being in the middle of the Pacific, maybe we should go for `enso=`, ie El Niño Southern Oscillation. The meaning *might* be obvious to our climate science community... =P)

Reply:
Loop as default is fine for me